event-sourcing/event-store-cassandra/src/main/scala/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStore.scala

```diff
@@ -38,7 +39,16 @@ class CassandraEventStore @Inject() (eventStoreDao: EventStoreDao) extends Event
 
   private def doAppendAll(events: Iterable[Event]): SMono[Void] = {
     Preconditions.checkArgument(Event.belongsToSameAggregate(events))
-    eventStoreDao.appendAll(events)
+
+    val preconditionPublisher = SFlux.fromIterable(events)
```

Comment:
This precondition for `CassandraEventStoreTest#appendShouldThrowWhenTryingToRewriteHistory` 
It will be more heavy when we don't have more table.
any better idea are welcome

Reply:
No IMO we should delete an entire aggregate. This can be done atomically. So we do not need preconditions.